encoding/decode_test.go

```diff
@@ -0,0 +1,50 @@
+package encoding
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/go-chat-bot/bot"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestDecode(t *testing.T) {
+	Convey("Decode", t, func() {
+		bot := &bot.Cmd{
+			Command: "decode",
+		}
+
+		Convey("Should decode a value", func() {
+			bot.Args = []string{"base64", "R28gaXMgYW4gb3BlbiBzb3VyY2UgcHJvZ3JhbW1pbmcgbGFuZ3VhZ2U="}
+			got, error := decode(bot)
+
+			want := "Go is an open source programming language"
+			So(error, ShouldBeNil)
+			So(got, ShouldEqual, want)
+		})
+
+		Convey("Should return a error message when pass a invalid hash", func() {
+			bot.Args = []string{"base64", "R28gaXMgYW4gb3BlbiBzb3VyY2Ugc", "HJvZ3JhbW1pbmcgbGFuZ3VhZ2U="}
+			got, error := decode(bot)
+
+			want := 0
+			So(error, ShouldBeNil)
+			So(strings.Index(got, "Error: "), ShouldEqual, want)
```

Comment:
In this case it would be better to use `So(got, ShouldStartWith, "Error: ")`

https://github.com/smartystreets/goconvey/wiki/Assertions#strings